File: 프로그래머스/unrated/181885. 할 일 목록/할 일 목록.js
function solution(todo_list, finished) {
  let index = []

  for (let i = 0; i < finished.length; i++) {
    if (!finished[i]) {
      index.push(i)
    }
  }

  let ans = index.map(v => {
    return todo_list[v]
  })

  return ans
}

File: 프로그래머스/0/181885. 할 일 목록/할 일 목록.java
import java.util.ArrayList;
import java.util.List;

class Solution {
  public String[] solution(String[] todo_list, boolean[] finished) {
    List<String> result = new ArrayList<>();

    for (int i = 0; i < finished.length; i++) {
      if (!finished[i]) result.add(todo_list[i]);
    }

    return result.toArray(new String[0]);
  }
}